planner/core/logical_plan_builder.go

```diff
@@ -3639,11 +3653,16 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, currentLev
 			b.ctx.GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack("We can only use the straight_join hint, when we use the leading hint and straight_join hint at the same time, all leading hints will be invalid"))
 		}
 	}
+	if orderedHJHintCnt > 1 {
+		orderedHashJoinTables = orderedHashJoinTables[:0]
+		b.ctx.GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack("We can only use one ordered_hash_join hint in one query block at most, when multiple ordered_hash_join hints are used, all ordered_hash_join hints will be invalid"))
```

Comment:
One block only one ordered hint?
For the SQL `from t1, t2, t3, t4 where ...`, we could use multiple hints in fact?

Reply:
+1.
And I find the logic is a bit strange here. What's the different between `ordered_hash_join(t1, t2)` and `ordered_hash_join(t1), ordered_hash_join(t2)`?

Reply:
> +1.
> And I find the logic is a bit strange here. What's the different between `ordered_hash_join(t1, t2)` and `ordered_hash_join(t1), ordered_hash_join(t2)`?

`ordered_hash_join(t1, t2)` means t1 is the build side, t2 is the probe side.
`ordered_hash_join(t1), ordered_hash_join(t2)` means t1 and t2 are both build side.

Reply:
Here is my mistake. Now, updated. PTAL @winoros  @time-and-fate 

Now, we have two parameters to describe the `ORDERED_HASH_JOIN` hint.
For example, `ordered_hash_join(t0, t1), ordered_hash_join(t2)`
The `buildSideHJTables` will contain `{t0, t2}`.
The `probeSideHJTables` will contain `{t1}`.

Reply:
will this be a little compilicated?

Reply:
For the hint of index join, the arguments are only considered to be the build side.
This hint is somewhat more complicated than that one.